test/framework/src/main/java/org/elasticsearch/test/hamcrest/ThrowableAssertions.java

```diff
@@ -25,6 +26,7 @@ public static void assertThatThrows(
     ) {
         try {
             code.run();
+            fail("Exception with type " + exceptionType.getName() + "should be thrown");
```

Comment:
Ah, this won't totally work because we catch `Throwable` below. We'll need to set a `success` flag here and then assert that `success == false` outside the `catch` block.

Reply:
ah, you are right. I'll update it and another same logci in `RepositoriesDuplicatedCreationIT`